java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -216,34 +142,53 @@ public ColumnVector(DType type, long rows, Optional<Long> nullCount,
    * @param contiguousBuffer the buffer that this is based off of.
    */
   private ColumnVector(long viewAddress, DeviceMemoryBuffer contiguousBuffer) {
+    super(viewAddress);
     offHeap = new OffHeapState(viewAddress, contiguousBuffer);
     MemoryCleaner.register(this, offHeap);
-    this.type = offHeap.getNativeType();
-    this.rows = offHeap.getNativeRowCount();
     // TODO we may want to ask for the null count anyways...
     this.nullCount = Optional.empty();
 
     this.refCount = 0;
     incRefCountInternal(true);
   }
 
-  static ColumnVector fromViewWithContiguousAllocation(long columnViewAddress, DeviceMemoryBuffer buffer) {
-    return new ColumnVector(columnViewAddress, buffer);
+  /**
+   * Retrieves the column_view for a cudf::column and if it fails to do so, the column is deleted
+   * and the exception is thrown to the caller.
+   * @param nativePointer the cudf::column handle
+   * @return the column_view handle
+   */
+  private static long getColumnViewFromColumn(long nativePointer) {
+    try {
+      return ColumnVector.getNativeColumnView(nativePointer);
+    } catch (CudfException ce) {
+      deleteCudfColumn(nativePointer);
+      throw ce;
+    }
   }
 
-  /**
-   * Returns a column of strings where, for each string row in the input,
-   * the first character after spaces is modified to upper-case,
-   * while all the remaining characters in a word are modified to lower-case.
-   *
-   * Any null string entries return corresponding null output column entries
+  /** Creates a ColumnVector from a column view handle
+   * @return a new ColumnVector
    */
-  public ColumnVector toTitle() {
-    assert type == DType.STRING;
-    return new ColumnVector(title(getNativeView()));
+  public static ColumnVector fromColumnView(ColumnView columnView) {
+    return new ColumnVector(ColumnView.copyColumnViewToCV(columnView.getNativeView()));
```

Comment:
I would expect `ColumnView` to have a non-static method that would copy itself into a `ColumnVector`, especially since this method is implemented completely in terms of the view.  It's reaching inside view instance and calling a static method that takes essentially all of the data represented by an instance.  That smells like a `ColumnView` member function to me.

If it does stay here, I'd prefer if it were more explicit in the name of the factory method that a copy is occurring.

Reply:
Sorry I must have misunderstood your comment from earlier on having a factory method. I have put this back in ColumnView as a member and renamed it as well just to be clear.